tensorflow/core/kernels/maxpooling_op_sycl.h

```diff
@@ -45,28 +45,28 @@ template <typename T>
 class MaxPool2DSYCL {
  public:
   static constexpr auto read_mode = cl::sycl::access::mode::read;
-  static constexpr auto write_mode = cl::sycl::access::mode::write;
+  static constexpr auto write_mode = cl::sycl::access::mode::discard_write;
```

Comment:
I am not 100% sure that we can use a discard_write in all cases but I think we should.

Reply:
Yeah I think we should be using discard_write in all these kernels.